packages/gui/src/hooks/useVerifyHash.ts

```diff
@@ -0,0 +1,147 @@
+import { useState, useEffect } from 'react';
+import isURL from 'validator/lib/isURL';
+
+import isContentHashValid from '../util/isContentHashValid';
+import getRemoteFileContent from '../util/getRemoteFileContent';
+import { MAX_FILE_SIZE } from './useVerifyURIHash';
+import { mimeTypeRegex } from '../util/utils.js';
+
+function isAudio(uri: string) {
+  return mimeTypeRegex(uri, /^audio/);
+}
+
+function isImage(uri: string) {
+  return mimeTypeRegex(uri, /^image/);
+}
+
+export default function useVerifyThumbnailHash(
+  uri: string,
+  metadataJson: any,
+  isPreview: boolean,
+  dataHash: string,
+): {
+  isValid: boolean;
+  isLoading: boolean;
+  error: string | undefined;
+  thumbnail: any;
+} {
+  const [isValid, setIsValid] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | undefined>();
+  const [thumbnail, setThumbnail] = useState({});
+
+  let lastError: any;
+
+  async function validateHash(metadataJson: any): Promise<void> {
+    let uris: string[] = [];
```

Comment:
this method is too complicated.
use one try { setLoading(true); ... everything else } catch { setError(error) }  finally { setLoading (false)}

instead of using setError many times just throw new Error()